gadgets/trace_tcp/program.bpf.c

```diff
@@ -0,0 +1,406 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2022 Microsoft Corporation
+//
+// Based on tcptracer(8) from BCC by Kinvolk GmbH and
+// tcpconnect(8) by Anton Protopopov
+#include <vmlinux.h>
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_endian.h>
+#include <gadget/mntns_filter.h>
+#include <gadget/types.h>
+#include <gadget/macros.h>
+#include <gadget/buffer.h>
+
+/* The maximum number of items in maps */
+#define MAX_ENTRIES 8192
+
+#define TASK_COMM_LEN 16
+
+enum event_type : u8 {
+	TCP_EVENT_TYPE_CONNECT,
+	TCP_EVENT_TYPE_ACCEPT,
+	TCP_EVENT_TYPE_CLOSE,
+};
+
+struct event {
+	struct gadget_l4endpoint_t src;
+	struct gadget_l4endpoint_t dst;
+
+	__u8 task[TASK_COMM_LEN];
+	gadget_mntns_id mntns_id;
+	gadget_timestamp timestamp;
+	__u64 ts_us;
```

Comment:
I think the format is the only difference. Both fields convey timestamp information. Are they both needed?

Reply:
I don't think so. We can remove this one.

Reply:
Removed it. Left TODO:
[ ] add `timestamp` column to the gadget output instead of `ts_us`. `timestamp` output will take more space on the screen though.

Reply:
I don't get it. What does your TODO refer to?

Reply:
For example in the output of the gadget you quoted [here](https://github.com/inspektor-gadget/inspektor-gadget/pull/2428#discussion_r1473171739), there is a `ts_us` column. Should we replace that field with timestamp?